From 464d07255f3ad28d7d3cb8cccbb905fbcce06c35 Mon Sep 17 00:00:00 2001
From: Rui Abreu Ferreira <raf-ep@gmx.com>
Date: Tue, 12 Jun 2012 11:13:25 +0100
Subject: [PATCH 08/11] Build clientserver branch on windows

- In windows we don't need our own clientserver implementation
  because the win32 vim already has one. But since but were active
  the tb-clientserver branch would not build in win32.
---
 src/CMakeLists.txt |  1 +
 src/gui_qt.cpp     | 12 ++++++++++++
 2 files changed, 13 insertions(+)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index b4daa08..b102fda 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -2,6 +2,7 @@ cmake_minimum_required(VERSION 2.8)
 
 find_package(Qt4 REQUIRED)
 set( QT_USE_QTGUI TRUE )
+set( QT_USE_QTNETWORK TRUE )
 include(${QT_USE_FILE} )
 
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
diff --git a/src/gui_qt.cpp b/src/gui_qt.cpp
index a881854..b1c7cf7 100644
--- a/src/gui_qt.cpp
+++ b/src/gui_qt.cpp
@@ -18,7 +18,14 @@
 #include "vimaction.h"
 #include "vimscrollbar.h"
 #include "fontdialog.h"
+
+//
+// Win32 can use its own clientserver
+// implementation
+//
+#if defined(FEAT_CLIENTSERVER) && !defined(QT_OS_WIN32)
 #include "commandserver.h"
+#endif
 
 extern "C" {
 
@@ -574,6 +581,8 @@ gui_mch_init()
 		gui_mch_enter_fullscreen();
 	}
 
+
+#if defined(FEAT_CLIENTSERVER) && !defined(QT_OS_WIN32)
 	//
 	// Client-Server
 	// - Start the server
@@ -594,6 +603,7 @@ gui_mch_init()
 		set_vim_var_string(VV_SEND_SERVER, serverName, -1);
 #endif
 	}
+#endif // FEAT_CLIENTSERVER
 
 	return OK;
 }
@@ -821,9 +831,11 @@ gui_mch_exit(int rc)
 	settings.setValue("state", window->saveState());
 	settings.endGroup();
 
+#if defined(FEAT_CLIENTSERVER) && !defined(QT_OS_WIN32)
 	// Close the Vim server
 	CommandServer *server = CommandServer::getInstance();
 	server->close();
+#endif // FEAT_CLIENTSERVER
 	QApplication::quit();
 }
 
-- 
2.3.1

